hashin.py

```diff
@@ -100,48 +102,24 @@ def run_single_package(spec, file, algorithm, python_versions=None, verbose=Fals
         package, version = spec, None
         # then the latest version is in the breadcrumb
 
-    data = get_package_data(package, verbose)
-    if not version:
-        version = get_latest_version(data)
-        assert version
-        if verbose:
-            _verbose("Latest version for", version)
-
-    # Independent of how you like to case type it, pick the correct
-    # name from the PyPI index.
-    package = data['info']['name']
-
-    try:
-        releases = data['releases'][version]
-    except KeyError:
-        raise PackageError('No data found for version {0}'.format(version))
-
-    if python_versions:
-        releases = filter_releases(releases, python_versions)
-
-    if not releases:
-        if python_versions:
-            raise PackageError(
-                "No releases could be found for {0} matching Python versions {1}"
-                .format(spec, python_versions)
-            )
-        else:
-            raise PackageError(
-                "No releases could be found for {0}"
-                .format(spec, python_versions)
-            )
-
-    add_hashes(releases, algorithm, verbose=verbose)
+    data = get_package_hashes(
+        package=package,
+        version=version,
+        verbose=verbose,
+        python_versions=python_versions,
+        algorithm=algorithm
+    )
```

Comment:
Seem here you're missing [these lines](https://github.com/peterbe/hashin/blob/6549e7a0cf88f62c5c38a2ac3a5b3ea069415f34/hashin.py#L104-L108)